### ISSUE TITLE
--ignore-underscore-args have no effect

### ISSUE THREAD

**Issue by finswimmer** [jsh9/pydoclint#216]
Hey,

given this example:

```py
def method(a: str, _b: str) -> None:
    """Doing stuff.

    :param a: some string
    """
    print(a, _b)
```

And running  `pydoclint --style=sphinx --arg-type-hints-in-docstring=False <path_to_file>`, I would expect `pydoclint` doesn't report any errors. Instead I'm getting:

```
1: DOC101: Function `method`: Docstring contains fewer arguments than in function signature.
1: DOC103: Function `method`: Docstring arguments are different from function arguments. (Or could be other formatting issues: https://jsh9.github.io/pydoclint/violation_codes.html#notes-on-doc103 ). Arguments in the function signature but not in the docstring: [_b: str].
```

On the given notes site I cannot find anything that might explains a formatting issue in the example. 

I also tried explicit setting `--ignore-underscore-args` to True, but without any success.

Any ideas?

Thanks a lot.

fin swimmer

**Comment by jsh9** [comment]
Hi @finswimmer ,

Thanks for raising this issue.

According to the code: https://github.com/jsh9/pydoclint/blob/b05a41ac5d300e95b893ba3ffdb2aa63a64c2f8b/pydoclint/visitor.py#L457-L464

"Underscore arguments" are arguments whose names contain only the `_` character.  I updated the documentation (https://github.com/jsh9/pydoclint/commit/299a57c94337d684280506a4287d173d76f273d50) to be more clear.

Do you have some use cases where arguments with leading `_` need to be ignored too?  If so, I can consider adding a new config option.

**Comment by finswimmer** [comment]
Hey @jsh9,

thanks a lot for  the clarification 👍 

> Do you have some use cases where arguments with leading _ need to be ignored too?

I'm coming from `darglint`. And this behaves like this. Is this enough use case? 😅 

I have several methods where I dependency inject things like a database connection. For those arguments I don't need/want a description in the docstring.

fin swimmer

**Comment by jsh9** [comment]
Yeah, I think a valid use case then.

I can make some code changes later this week.

**Comment by finswimmer** [comment]
This sounds great 🚀

**Comment by jsh9** [comment]
The new feature is published in 0.6.3.

There's now a new config option: `--ignore-private-args`: https://jsh9.github.io/pydoclint/config_options.html#14---ignore-private-args-shortform--ipa-default-false

**Comment by finswimmer** [comment]
Works great 👍  Thanks a lot!

**Comment by jonyscathe** [comment]
The documentation states that this new --ignore-private-args defaults to False, however I have to set `ignore-private-args = false` in my `[tool.flake8]` section to order to keep the previous behavour when running with flake8.
Is this expected?

**Comment by jsh9** [comment]
Hi @jonyscathe , it's a typo on my end.  I'll fix it.

**Comment by jsh9** [comment]
The fix is published in 0.6.5.
